pkg/policy/selectorcache.go

```diff
@@ -474,18 +472,11 @@ func (sc *SelectorCache) UpdateIdentities(added, deleted identity.IdentityMap, w
 	}
 }
 
-// GetNetsLocked returns the most specific CIDR for an identity. For the "World" identity
-// it returns both IPv4 and IPv6.
-func (sc *SelectorCache) GetNetsLocked(id identity.NumericIdentity) []*net.IPNet {
+// GetPrefixLocked returns the most specific CIDR for an identity, if any.
+func (sc *SelectorCache) GetPrefixLocked(id identity.NumericIdentity) netip.Prefix {
```

Comment:
What if we make a separate `sync.Map[id]prefix`. Then we can drop this lock, which would be a big help while rolling out incremental updates.

Reply:
Good point, this would remove the need for the selector cache locking added here.

Reply:
Done :-)

Reply:
Do you think we could also drop the locking here:
```
// ConsumeMapChanges transfers the changes from MapChanges to the caller,
// locking the selector cache to make sure concurrent identity updates
// have completed.
// PolicyOwner (aka Endpoint) is also locked during this call.
func (p *EndpointPolicy) ConsumeMapChanges() (adds, deletes Keys) {
	p.selectorPolicy.SelectorCache.mutex.RLock()
	defer p.selectorPolicy.SelectorCache.mutex.RUnlock()
```
From the comment it seems possible that the mutex is needed also to make sure an identity that is being added is actually done before we proceed here. So while `sync.Map()` is safe, if we drop this lock it could happen that the prefix is not yet found when we ask for it?

Or do we have some other synchronization that takes care of the ordering of events here?

Reply:
Note that after the last commit the access pattern to `SelectorCache.GetPrefix()` is different in that it is called for the added entry only, as everything else is already in the index.

For incremental changes it is the new identity that was just added that is of interest, though.

Reply:
It seems the Prefix is now added to the `prefixMap` strictly before incremental updates are created, but maybe we need to be synchronizing for the concurrent updates to have propagated to the MapChanges so that we get them? Without this lock it could be possible that we "used" the trigger too early and just barely missed some of the updates, and we're not going to be triggered again until some periodic map sync etc.?

Reply:
We are explicitly handing this part through a waitgroup before we call endpoint's ApplyPolicyMapChanges:
```
			notifyWg.Wait()
			if err := ep.ApplyPolicyMapChanges(proxyWaitGroup); err != nil && !errors.Is(err, endpoint.ErrNotAlive) {
```

Given this, I'll remove this lock as well!